config/routes.rb

```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  get '/users/:user_id/posts' , to: 'posts#index'
+  get '/users/:user_id/posts/:id' , to: 'posts#show'
+  get '/users' , to: 'users#index'
+  get '/users/:id' , to: 'users#show'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
```

Comment:
- [ ] Good job defining routes for accessing user profiles👏👏.  However, you might want to consider using the resourceful routes provided by Rails, such as `resources :users` and `resources :posts`, which can generate these routes more conventionally and efficiently. 👍